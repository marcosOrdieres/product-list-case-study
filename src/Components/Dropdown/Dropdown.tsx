import React from 'react';
import styled from "styled-components";

export const DropdownWrapper = styled.form`
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items:center;
`;

export const StyledSelect = styled.select`
  height: 100%;
  padding: 0.5rem;
  margin-bottom: 1rem;
`;

export const StyledLabel = styled.label`
  margin-bottom: 1rem;
  font-size:15px;
  color:#3399ff
`;

export interface DropdownProps {
  formLabel?: string,
  action?: string,
  onChange?: any,
  children?: JSX.Element[],
};

const Dropdown = (props: DropdownProps) => (
  <DropdownWrapper
    action={props.action}
    onChange={props.onChange}
  >
    <StyledLabel htmlFor="services">
      {props.formLabel}
    </StyledLabel>
    <StyledSelect id="services" name="services">
      {props.children}
    </StyledSelect>
    {/* <StyledButton type="submit" value={props.buttonText} /> */}
  </DropdownWrapper>
);

export default Dropdown;