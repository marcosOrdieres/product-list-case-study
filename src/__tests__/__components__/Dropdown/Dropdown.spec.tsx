import React from 'react';
import { render } from '@testing-library/react';
import Dropdown from '../../../Components/Dropdown';


describe('Dropdown', () => {
  it('should render Dropdown and match styles', () => {
    const { asFragment } = render(
      <Dropdown formLabel='' action='' onChange={() => 'changed'}>
        <div />
        <div />
      </Dropdown>

    );
    expect(asFragment()).toMatchSnapshot();
  });
});