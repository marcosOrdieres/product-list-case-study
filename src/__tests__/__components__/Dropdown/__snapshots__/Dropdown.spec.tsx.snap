// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render Dropdown and match styles 1`] = `
<DocumentFragment>
  <form
    action=""
    class="sc-bdfBQB ncswd"
  >
    <label
      class="sc-dlfnuX dbiQCg"
      for="services"
    />
    <select
      class="sc-gsTEea bCJcBN"
      id="services"
      name="services"
    />
  </form>
</DocumentFragment>
`;
